runtime/src/chplmemtrack.c

```diff
@@ -98,7 +99,19 @@ static size_t totalAllocated = 0; /* total memory allocated */
 static size_t totalFreed = 0;     /* total memory freed */
 static size_t totalEntries = 0;     /* number of entries in hash table */
 
-static chpl_sync_aux_t memTrack_sync;
+
```

Comment:
Add a comment about why we use a pthread_mutex (and why it's safe here)?

Reply:
WIll do.